tests/frontend/test_configuration.py

```diff
@@ -46,107 +46,360 @@
         "authentication_token": "071cdcce-9241-4965-93af-4a4dbc739135",
         "hostname": "localhost",
         "use_ssl": True,
-        "debug": False,
         "port": 443,
+        "debug": False,
     }
 }
 
+OTHER_EXPECTED_CONFIG = {
+    "api": {
+        "authentication_token": "SomeAuth",
+        "hostname": "SomeHost",
+        "use_ssl": False,
+        "port": 56,
+        "debug": True,
+    }
+}
 
-class TestConfiguration:
-    """Tests for the configuration class"""
+environment_variables = [
+                    "SF_API_AUTHENTICATION_TOKEN",
+                    "SF_API_HOSTNAME",
+                    "SF_API_USE_SSL",
+                    "SF_API_DEBUG",
+                    "SF_API_PORT"
+                    ]
+
+def tear_down_all_env_var_defs():
+    """Making sure that no environment variables are defined after."""
+    for key in environment_variables:
+        if key in os.environ:
+            del os.environ[key]
+            assert key not in os.environ
```

Comment:
This is neat, but I would recommend using the pytest `monkeypatch.setenv()` and `monkeypatch.delenv()` inside the tests, rather than doing it by hand

Reply:
Great, thanks for that!